src/commons/eventsourcing/kv.rs

```diff
@@ -207,9 +221,14 @@ impl KeyValueStoreDiskImpl {
     }
 
     fn store<V: Any + Serialize>(&self, key: &KeyStoreKey, value: &V) -> Result<(), KeyValueError> {
-        let mut f = file::create_file_with_path(&self.file_path(key))?;
+        let swap_file_path = self.swap_file_path(key);
+        let file_path = self.file_path(key);
+        let mut swap_file = file::create_file_with_path(&swap_file_path)?;
         let json = serde_json::to_string_pretty(value)?;
-        f.write_all(json.as_ref())?;
+        swap_file.write_all(json.as_ref())?;
+
+        fs::rename(swap_file_path, file_path)?;
```

Comment:
Is rename safe at this point given that `swap_file` hasn't yet gone out of scope and so the file is not yet closed? Perhaps it works anyway on Linux as the writing happens to an FD which is unaffected by a rename?